src/MathOptInterface.jl

```diff
@@ -30,9 +30,44 @@ abstract type AbstractOptimizer <: ModelLike end
     optimize!(optimizer::AbstractOptimizer)
 
 Start the solution procedure.
+
+    optimize!(
+        dest::AbstractOptimizer,
+        src::ModelLike,
+    )::IndexMap
+
+A single call similar to calling [`copy_to(dest, src)`](@ref `copy_to`) followed
+by `optimize!(dest)`. Return an [`IndexMap`](@ref) and a `Bool` `copied`.
+The [`IndexMap`](@ref) object translates variable and constraint
+indices from the `src` model to the corresponding indices in the `dest`
+optimizer, similarly to the [`IndexMap`](@ref) returned by [`copy_to`](@ref).
+If `copied` is `true`, `src` was copied to `dest` so `optimize!(dest)`
+can be called again on `dest`, e.g., after changing the starting values
+or optimizer attributes. On the other hand, if `copied` is `false`, the
+model was not copied to `dest` so calling `optimize!(dest)` is not allowed.
+However, attributes for which [`is_set_by_optimize`](@ref) is true can be
+queried from `dest`.
+
+Subtypes, say `Optimizer`, of [`AbstractOptimizer`](@ref) should
+* either implement [`MOI.copy_to(::Optimizer, ::MOI.ModelLike)`](@ref `copy_to`) and
+  `optimize!(::Optimizer)` or
+* implement `optimize!(::Optimizer, ::MOI.ModelLike)`.
```

Comment:
This is a bit nasty. We now have
 * The incremental interface
 * `copy_to` then single-argument `optimize!`
 * Two-argument `optimize!`

Do we need `supports_copy_to_interface` like `supports_incremental_interface`?

Reply:
What would this be useful for? The supports_incremental_interface is there for instantiate

Reply:
How does a user know to call `optimize!(::AbstractOptimizer)` or `optimize!(::AbstractOptimizer, ::ModelLike)`? Or should they only ever call `optimize!(::AbstractOptimizer)` and use a caching optimizer? 

If this is only ever used by the `CachingOptimizer`, it should almost be a `Utilities.optimize!` method.

If you had `supports_`, you could do this:
```julia
if MOI.supports_copy_to_interface(m.optimizer)
    indexmap = MOI.copy_to(m.optimizer, m.model_cache)
    MOI.optimize!(m.optimizer)
    m.state = ATTACHED_OPTIMIZER
else
    indexmap = MOI.optimize!(m.optimizer, m.model_cache)
    m.state = EMPTY_OPTIMIZER
end
m.model_to_optimizer_map = indexmap
m.optimizer_to_model_map = _reverse_index_map(index_map)
```